hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -253,6 +254,12 @@ def from_typing_type(thing):
             for T in list(union_elems) + [elem_type]
         ):
             mapping.pop(typing.ByteString, None)
+    elif (
+        (not mapping)
+        and isinstance(thing, getattr(typing, "ForwardRef", ()))  # missing from py36
```

Comment:
We have a `compat` helper for `ForwardRef`: https://github.com/HypothesisWorks/hypothesis/blob/f5289ac715e9c54072727bc4226b2eded598bbcd/hypothesis-python/src/hypothesis/internal/compat.py#L69-L80 Maybe it is better to use it instead?